### PR TITLE
Add dependabot config

### LINKED ISSUE #371: Configure Dependabot using a YAML config file
## Description

For this repository, Dependabot is enabled but not configured. Adding a `.github/dependabot.yml` file should fix that. Using the file we can configure things like the update schedule and the labels to attach to the PRs made by Dependabot.

## Expectation

Dependabot should make regular, well-labeled PRs to the repository.

## Additional context

You can use the [config from the API repo](https://github.com/WordPress/openverse-api/blob/b760ca1c403265bafffda5b831033186b3d0c50f/.github/dependabot.yml) as a reference. Also see the [GitHub docs](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#about-the-dependabotyml-file) describing this config file.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #371 by @zackkrida

## Description

This PR adds a dependabot configuration to the repository.

Based on my understanding of the [dependabot documentation](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file), I believe dependabot will only try to update _pinned dependencies_. We only pin dependencies in cases where the dependency itself is not defined in the [Airflow constraints file](https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#constraints-files). As such I think it should be safe to have dependabot manage the _pinned_ dependencies, since those ones aren't managed by Airflow! I'm not sure how best to test this, we may just need to merge it and find out :smile: Dependabot should be able to handle the 3 different `requirements_*.txt` files we have.

I also renamed one of our files from `requirements.txt` to `requirements_tooling.txt` to make its purpose clearer.

## Testing Instructions

1. Try `just install` locally and make sure it still works

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by rwidom — APPROVED** [review]
I'm not 💯 confident what "pinned" means here, but I'm assuming that it means when there is a specific version listed in the requirements file. I did try `just install` with this branch and it worked fine. 👍 Just one tiny comment.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# current Github-native version of Dependabot
+version: 2
+
+updates:
+  # Enable version updates for Python libraries in `requirements_dev.txt`
+  # These can be updated more frequently than those in `requirements_prod.txt` since
+  # they are not pinned by the Airflow constraints file.
+  - package-ecosystem: pip
+    directory: /
+    # Check for updates once a week
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply .github/dependabot.yml:14]: Oops! Thanks, I corrected this in https://github.com/WordPress/openverse-api/pull/941 but forgot to do so here 😅

**Comment by AetherUnbound** [comment]
> I'm not 100 confident what "pinned" means here, but I'm assuming that it means when there is a specific version listed in the requirements file.

That's exactly it! Unpinned dependencies are those just listed in the file without a version attached to them, meaning that whatever the latest version is (or whatever the constraints dictate) is what gets installed.

**Review by zackkrida — APPROVED** [review]
lgtm!

**Review by stacimc — APPROVED** [review]
Reasoning seems sound to me, this looks good!

**Pull request merged**